x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/XPackSettingsTests.java

```diff
@@ -133,6 +135,48 @@ public void testRemoteClusterSslSettings() {
         ).forEach(key -> assertThat(remoteClusterSslSettingKeys, not(hasItem(key))));
     }
 
+    public void testSecurityMustBeEnableForRemoteClusterServer() {
+        // Security on, remote cluster server off
+        final Settings.Builder builder1 = Settings.builder();
+        if (randomBoolean()) {
+            builder1.put("xpack.security.enabled", "true");
+        }
+        if (randomBoolean()) {
+            builder1.put("remote_cluster_server.enabled", "false");
+        }
+        assertThat(XPackSettings.SECURITY_ENABLED.get(builder1.build()), is(true));
+
+        // Security off, remote cluster server off
+        final Settings.Builder builder2 = Settings.builder().put("xpack.security.enabled", "false");
+        if (randomBoolean()) {
+            builder2.put("remote_cluster_server.enabled", "false");
+        }
+        assertThat(XPackSettings.SECURITY_ENABLED.get(builder2.build()), is(false));
+
+        // Security on, remote cluster server on
+        final Settings.Builder builder3 = Settings.builder().put("remote_cluster_server.enabled", "true");
+        if (randomBoolean()) {
+            builder3.put("xpack.security.enabled", "true");
+        }
+        final Settings settings3 = builder3.build();
+        assertThat(XPackSettings.SECURITY_ENABLED.get(settings3), is(true));
+        assertThat(RemoteClusterPortSettings.REMOTE_CLUSTER_SERVER_ENABLED.get(settings3), is(true));
+
+        // Security off, remote cluster server on
+        final Settings settings4 = Settings.builder()
+            .put("remote_cluster_server.enabled", "true")
+            .put("xpack.security.enabled", "false")
+            .build();
+        final IllegalArgumentException e4 = expectThrows(
+            IllegalArgumentException.class,
+            () -> XPackSettings.SECURITY_ENABLED.get(settings4)
+        );
+        assertThat(
+            e4.getMessage(),
+            containsString("Security [xpack.security.enabled] must be enabled to use the remote cluster server feature")
+        );
```

Comment:
I think the unit test is sufficient to show that node will not start when settings are incompatible. But please let me know if you prefer to have an integration test for the effect.